server/src/main/scala/passengersplits/core/PassengerInfoRouterActor.scala

```diff
@@ -72,52 +72,61 @@ trait SimpleRouterActor[C <: Actor] {
   }
 }
 
-class PassengerSplitsInfoByPortRouter extends PersistentActor with PassengerQueueCalculator with ActorLogging
-  with SimpleRouterActor[PassengerInfoRouterActor] {
+class PassengerSplitsInfoByPortRouter extends PersistentActor with PassengerQueueCalculator with ActorLogging {
 
-  var latestFileName: Option[String] = None
+  case class State(latestFileName: Option[String],
+                   flightManifests: Map[String, VoyageManifest])
+
+  var state = State(None, Map.empty)
+  val dcPaxIncPercentThreshold = 50
+
+  implicit def implLog = log
 
-  def childProps = Props[PassengerInfoRouterActor]
+  def manifestKey(vm: VoyageManifest) = s"${vm.ArrivalPortCode}-${padTo4Digits(vm.VoyageNumber)}@${SDate.jodaSDateToIsoString(vm.scheduleArrivalDateTime.get)}"
 
   override def receiveCommand = LoggingReceive {
     case ManifestZipFileInit =>
       log.info(s"PassengerSplitsInfoByPortRouter received FlightPaxSplitBatchInit")
       sender ! PassengerSplitsAck
 
     case manifest: VoyageManifest =>
-      log.info(s"telling children about ${manifest.summary}")
-      sendToChild(manifest)
       persist(voyageManifestToMessage(manifest)) { manifestMessage =>
         log.info(s"API: saving ${manifest.summary}")
         context.system.eventStream.publish(manifestMessage)
       }
+      if (SingleFlightActor.shouldAcceptNewManifest(manifest, state.flightManifests.get(manifestKey(manifest)), dcPaxIncPercentThreshold)) {
+        addManifest(manifest)
+      }
+      sender ! PassengerSplitsAck
+
     case report: ReportVoyagePaxSplit =>
       val replyTo = sender
-      val handyName = s"${report.destinationPort}/${report.voyageNumber}@${report.scheduledArrivalDateTime.toString}"
-
-      val name = childName(report.destinationPort)
-      log.info(s"$replyTo asked for us to look for ${report} in $name, $handyName")
-      val child = childActorMap get (name)
-      child match {
-        case Some(c) =>
-          log.debug(s"Child singleflight actor found child $name $handyName ")
-          c.tell(report, replyTo)
-        case None =>
-          log.debug(s"Child singleflight actor doesn't exist yet $name $handyName ")
-          replyTo ! FlightNotFound(report.carrierCode, report.voyageNumber, report.scheduledArrivalDateTime)
+      Future {
+        val key = s"${report.destinationPort}-${padTo4Digits(report.voyageNumber)}@${SDate.jodaSDateToIsoString(report.scheduledArrivalDateTime)}"
```

Comment:
extract this key construction (and the other key construction) to functions that sit near each other